pkg/minishift/config/featuretoggle.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+var FeatureToggles *FeatureTogglesType
+
+type FeatureTogglesType struct {
```

Comment:
what do you need this type really for?

Reply:
Would it not be enough to have a exported bool flag - _minishiftConfig.EnableExperimental_?

Reply:
If a new type, why not just _FeatureToggle_? I am not a big fan of 'Type' as part of the struct name.

Reply:
As `var FeatureToggles *FeatureToggles` is not possible. The idea would be to also allow this to be used for #1293, which `FeaturesToggles.HyperVEnabled` or similar. At the moment we can not avoid that the static IP address is a solution that can only work as intended by pre-provisioning on HyperV. However I do not think littering the code with `if HostConfig.vmDriver == "hyperv"` is a good idea.

Reply:
Ok, so _FeatureTogglesType_ is a container for experimental features, like in #1293. What's about the service-catalog? I would have thought that this would be part of this refactoring as well. e.g. _FeaturesToggles.ServiceCatalogEnabled_. 

BTW, you could call the struct _FeatureToggle_ or _FeatureToggleGroup_ or _FestureToggleList_

Reply:
@gbraad Are your working on it? or Keeping it struct as it is?

Reply:
#1316 will use this